[WIP] Improve headers in EC-Lab output

dalg24 opened the pull request:
Addressing Issue #86

Commit:
add method inspect() to pycap.EnergyStorageDevice

Commit:
added DefaultInspector to extract the data

``inspect()`` method takes reference to an EnergyStorageDevice
cannot be const unless we change EnergyStorageDevice class

also removed unused ``compute_equivalent_circuit`` from python wrappers

Commit:
make postprocessor compute specific surface area

Commit:
update test device wrappers

dalg24 commented:
Note: we will also need to provide access to the post processor in the new supercap

Commit:
building headers for EC-Lab ascii file

Commit:
inspector gather info needed by headers

dalg24 commented:
I deferred the responsibility to gather all the info displayed in the EC-Lab headers to the inspector.
Surprisingly, the input parameters used to build the supercap seem to be the most delicate to get.

Commit:
save copy ptree used to construct supercap

Commit:
convert to correct units in EC-Lab headers

dalg24 commented:
Now all the hooks are in place. We need to compute correctly the mass of the electrode (computing total mass right now) and decide whether we print info for each electrode or not.

Commit:
fix test energ storage

Rombur commented on cpp/source/default_inspector.cc:
```diff
@@ -0,0 +1,38 @@
+
+void DefaultInspector::inspect(EnergyStorageDevice *device)
+{
+    auto super_capacitor = dynamic_cast<SuperCapacitor<2>*>(device);
```
This won't work in 3D.

Rombur commented on cpp/source/supercapacitor.h:
```diff
@@ -79,6 +79,16 @@ class SuperCapacitor : public EnergyStorageDevice
+  /**
+  * Granting access to the post processor for the inspector
+  */
+  std::shared_ptr<Postprocessor<dim>> get_post_processor() const;
```
Add this to `New_SuperCapacitor` also. It will be easier when we remove this class.

Commit:
removed template EC-Lab ascii file

Commit:
treat anode and cathode separetaly

I had trouble with the source encoding with Python2.7
I had to defined the encoding for the file impedance_specroscopy.py

Commit:
make default inspector work for dim=3

Rombur commented on cpp/source/default_inspector.cc:
```diff
@@ -0,0 +1,68 @@
+            ptree->get<double>("geometry.anode_electrode_thickness");
+        data["cathode_electrode_thickness"] =
+            ptree->get<double>("geometry.cathode_electrode_thickness");
+        // TODO
```
What's left todo?

dalg24 replied:
Nothing I forgot to remove it

Rombur commented on python/source/impedance_spectroscopy.py:
```diff
@@ -1,3 +1,5 @@
+# This Python file uses the following encoding: iso-8859-1
+#
```
Move this after the copyright notice.

dalg24 replied:
No it needs to be within the first two lines. See https://www.python.org/dev/peps/pep-0263/

Rombur replied:
OK

Commit:
removed outdated TODO comment
